src/containerapp/azext_containerapp/custom.py

```diff
@@ -1264,14 +1277,17 @@ def containerapp_up(cmd,
     extension.create_if_needed()
     custom_location.create_if_needed()
     env.create_if_needed(name)
-
     if source or repo:
         if not registry_server:
             _get_registry_from_app(app, source)  # if the app exists, get the registry
-        _get_registry_details(cmd, app, source)  # fetch ACR creds from arguments registry arguments
+        if source:
+            _get_registry_details_without_get_creds(cmd, app, source)
+        if repo:
```

Comment:
what if both source and repo provided?

Reply:
This is not allowed:
When input --source --repo together, throw:
Cannot use --source and --repo together. Can either deploy from a local directory or a Github repo

code in [_validate_up_args](https://github.com/Azure/azure-cli-extensions/blob/201ea317d5103eeaf165c7b2564be5f8c0e4c377/src/containerapp/azext_containerapp/_up_utils.py#L1034)